apicast/src/provider.lua

```diff
@@ -385,8 +385,15 @@ function _M.access(service)
     ngx.var.cached_key = concat({service.id, params.app_id, params.app_key}, ':')
 
   elseif backend_version == 'oauth' then
-    ngx.var.access_token = parameters.access_token
-    params.access_token = parameters.access_token
+    if credentials.location == 'headers' then
```

Comment:
Originally I was thinking about checking the `Authorization` header for all modes.
Btw, not just `Bearer`, but also `Basic` and possibly `Digest`. That is long overdue feature request.

But maybe that could break someone's use (If they use Authorisation for something else already) so it should be opt it. 

To fix this properly, we should have another option in the UI to use Authorization header (instead of params/header). Right? That could take time.

So what if we make new feature behind a flag (like the path routing), that uses Authorisation header first, before taking look in the configured method. Behind a flag it could be used for all 3 auth methods until we make UI option to make it explicit.